[HEEaZ] iP

HEEaZ opened the pull request:
### LukePRO

> "If you can dream it, you can do it" - [Walt Disney](https://thewaltdisneycompany.com/)

LukePRO frees some of your mind to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUpER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/HEEaZ/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😸 

and its **FREE**

Features:
- [x] Managing Tasks
- [ ] Managing Deadlines (coming soon)
- [ ] Reminders (coming soon)

If you java programmer, you can use it to practice java too!. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

HEEaZ committed:
Changed logo name to Luke

HEEaZ committed:
Add greeting and bye msg

HEEaZ committed:
Added echo functionality

HEEaZ committed:
Add add, list functionality

HEEaZ committed:
Add marking feature

HEEaZ committed:
Create Task subclasses

HEEaZ committed:
Add todo, deadline, and event functionality

HEEaZ committed:
Add testing

HEEaZ committed:
Refactor and Add exception handling

HEEaZ committed:
Add delete functionality

HEEaZ committed:
minor refactor

HEEaZ committed:
Update gitignore to ignore data files

HEEaZ committed:
Create save functionality

HEEaZ committed:
Fix minor display inconsistency

HEEaZ committed:
Modify testfile to run from ip workspace

HEEaZ committed:
modify .gitignore to only exclude savedTasks txtfile

HEEaZ committed:
Create default saved file

HEEaZ committed:
Update test to use default saveTasks file

HEEaZ committed:
Update .gitignore to ignore actual savefile

HEEaZ committed:
Update test to output actual savefile

HEEaZ committed:
Update tests

HEEaZ committed:
Merge branch 'branch-Level-7'

HEEaZ committed:
Update Deadline to use LocalDate

HEEaZ committed:
Update savedTaskDefaults

HEEaZ committed:
Update tests

HEEaZ committed:
Merge branch 'branch-Level-8'

HEEaZ committed:
Change Util to Ui

HEEaZ committed:
Update .gitignore to ignore garbage tmp data file

HEEaZ committed:
OOP Refactoring

HEEaZ committed:
Merge branch 'branch-A-MoreOOP'

HEEaZ committed:
Refactor scanner into Ui

HEEaZ committed:
remove excess imports

HEEaZ committed:
Move files into duke package

HEEaZ committed:
Merge branch 'add-gradle-support'

HEEaZ committed:
Add checkstyle to gradle

HEEaZ committed:
Fix switch indentation

HEEaZ committed:
Create equals methods for Tasks

HEEaZ committed:
Change error message for Tasklist.add

HEEaZ committed:
Create test for getAll, add, delete in TaskList

HEEaZ committed:
Add Javadocs

HEEaZ committed:
Fix styling issues (less javadocs)

HEEaZ committed:
Update runtest.sh to compile from correct folder

HEEaZ committed:
Add find functionality

HEEaZ committed:
Revamp list and find logic and process

HEEaZ committed:
Merge branch 'branch-A-JavaDoc'

HEEaZ committed:
Make constants screaming snake case

HEEaZ committed:
Fix setting boolean method naming

HEEaZ committed:
Merge branch 'branch-A-CodingStandard'

HEEaZ committed:
Merge branch 'branch-Level-9'

cyaoxuan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+            tasks = new TaskList();
+        }
+    }
+    public static void main(String[] args) {
```
Should there be a space before the function?

cyaoxuan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+        case "todo":
+            // Fallthrough
+        case "deadline":
+            // Fallthrough
```
Good inclusion of the comments, and following the coding standard regarding switch statements!

cyaoxuan commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,102 @@
+        super(name);
+        this.finishByTime = finishByTime;
+    }
+    Deadline(boolean isDone, String name, LocalDate finishByTime) {
```
Should there be a space before the function?

cyaoxuan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,82 @@
+        }
+    }
+
+    public ArrayList<Task> load() throws LukeException {
```
Perhaps this function could have a more descriptive name? For example loadFromTaskFile

cyaoxuan reviewed (commented):
Your code is really clean and easy to read! A small improvement I would suggest is to have more consistent empty lines between functions or logical blocks. Good job :D

HEEaZ committed:
Implement GUI

HEEaZ committed:
Add gui save and exit functionality

HEEaZ committed:
Add greeting

rayshawntan reviewed (commented):
Overall, I think your code looks good. However, you seem to be missing Javadoc header for your classes at the top and your constructors.

HEEaZ committed:
modify build.gradle to use launcher

HEEaZ committed:
Add assert statements to document assumptions

The code makes some assumptions at various places.

This might make regressions and bugs harder to spot as we continue
to develop the code.

Let's include assertion checks at these places where assumptions are
made.

This allows us to check assumptions made in the code and ensure that
such assumptions are still being upheld, even as we change the code.

HEEaZ committed:
Merge pull request #1 from HEEaZ/branch-A-Assertions

Add assertions to document assumptions

HEEaZ committed:
Remove unused method Ui.list

HEEaZ committed:
Extract createTask method from Storage.load

Storage::load has a long and complicated method body

This affects the readability of the code.

Let's extract out the portion of code relating to creating the task into
a new function, createTask.

Let's further extract the portion of the code that deals with
determining if the task has been done into a new function, getTaskStatus.

This reduces the size of each method, which makes the code easier to
read.

HEEaZ committed:
Merge pull request #2 from HEEaZ/branch-A-CodeQuality

Improve code quality

HEEaZ committed:
Add CI

HEEaZ committed:
Fix checkstyle errors

HEEaZ committed:
Merge pull request #3 from HEEaZ/branch-A-CI

Add CI to project

HEEaZ committed:
Remove IO redirection test from github actions

HEEaZ committed:
Modify commands to allow shorthands t, d, e

HEEaZ committed:
Modify commands to allow shorthands ls, m, um, del, f

HEEaZ committed:
Fix shorthand bugs

HEEaZ committed:
Merge pull request #4 from HEEaZ/branch-C-FriendlierSyntax

Add friendlier syntax

HEEaZ committed:
Make images circular

HEEaZ committed:
Fix list getting cut off

HEEaZ committed:
Merge branch 'branch-A-BetterGui'

HEEaZ committed:
Add ui screenshot

HEEaZ committed:
Create user guide

HEEaZ committed:
Merge branch 'branch-A-UserGuide'

HEEaZ committed:
Add Chatbot Name to title

HEEaZ committed:
Fix errors flagged by se-bot

HEEaZ committed:
Fix saving without a data directory

HEEaZ committed:
Fix checkstyle errors

HEEaZ committed:
Update README.md
